test/cspviolationdetector/DefaultCSPViolationDetector.test.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import * as chai from 'chai';
+
+import DefaultCSPViolationDetector from '../../src/cspviolationdetector/DefaultCSPViolationDetector';
+import DOMMockBuilder from '../dommock/DOMMockBuilder';
+
+describe('DefaultCSPViolationDetector', () => {
+  let expect: Chai.ExpectStatic;
+  let domMockBuilder: DOMMockBuilder | null = null;
+
+  before(() => {
+    domMockBuilder = new DOMMockBuilder();
+    expect = chai.expect;
+  });
+
+  afterEach(() => {
+    if (domMockBuilder) {
+      domMockBuilder.cleanup();
+      domMockBuilder = null;
+    }
+  });
+
+  describe('construction', () => {
+    it('can be constructed', () => {
```

Comment:
You need a test to make sure that having two detectors doesn't cause two log lines.

Reply:
Sure, will add.